tests/codebuild/testfiles/failing-xgboost-mnist-hpo.yaml

```diff
@@ -0,0 +1,116 @@
+# This HPO job was designed to fail during the "Training" phase.
+# The image specified refers to a plain Tensorflow DLC image that does not have
+# the correct entrypoint for a SageMaker training job. This will fail with a 
+# python exception that should be captured by the operator.
```

Comment:
Will this generate a `FailureReason` for the HPO parent job, or only for the children TrainingJobs? We should have integration test coverage for HPO failure reason if it's at all feasible.

Reply:
If all child training jobs fail within an HPO job, the parent HPO job is given the `FailureReason` "All training jobs failed. Please take a look at the training jobs failures to get more details.". Since we are using the same faulty training container within this example, this test will always fail all 2/2.